docs/components/properties.md

```diff
@@ -567,6 +567,61 @@ and TSX:
 <todo-list-item isComplete={false}></todo-list-item>
 ```
 
+When using union types, the type of a component's `@Prop()` value can be ambiguous at runtime.
+In the provided example, under what circumstances does `@Prop() isComplete` function as a `string`, and when does it serve as a `boolean`?
+
+When using a component in HTML, the runtime value of a `@Prop()` is always a string.
+This is a result of setting the HTML attribute for the custom element.
+```html
+<!-- Since this is HTML, the value of `isComplete` in `ToDoListItem` will be a string -->
+
+<!-- Set isComplete to "true". -->
+<todo-list-item is-complete="true"></todo-list-item>
+<!-- Set isComplete to 'false' -->
+<todo-list-item is-complete="false"></todo-list-item>
+```
+
+Likewise, when the attribute on a component is set, the runtime value of a `@Prop()` is always a string.
+```html
+<script>
+  // both of these `setAttribute` calls set the property `isComplete` to "true" (string)
+  document.querySelector('todo-list-item').setAttribute('is-complete', true);
+  document.querySelector('todo-list-item').setAttribute('is-complete', "true");
+  // both of these `setAttribute` calls set the property `isComplete` to "false" (string)
+  document.querySelector('todo-list-item').setAttribute('is-complete', false);
+  document.querySelector('todo-list-item').setAttribute('is-complete', "false");
+</script>
+```
+
+However, if the property of a custom element is directly changed, the `typeof` of the `@Prop()` provided is used.
```

Comment:
```suggestion
However, if the property of a custom element is directly changed by accessing the setter on the element, the supplied value is used without performing any type conversion. In other words, the `typeof` the value will remain the same.
```

Possibly change, just to break down a bit what is going on (getting the value in via the setter / getter or defined property as opposed to the attribute)

Reply:
I think this might be a little too in the weeds for folks - what do you think about something like 6648bdc5a? I tried to simplify what I had previously here to try say "What you give is what you get" 😆

Reply:
sure, I think that's fine, just thinking we want to clarify we're talking about runtime types which means possible type conversion